### ISSUE TITLE
Deploy fails for LeBlender editors, when radiobuttons hasn't select a prevalue

### ISSUE THREAD

**Issue by bjarnef** [umbraco/Umbraco.Courier.Contrib#20]
When I in my frontpage grid have different LeBlender grid editors, which have radiobuttons as properties, it seems to fail to deploy, when no value for these are selected.

![image](https://cloud.githubusercontent.com/assets/2919859/22407624/659e1968-e66a-11e6-9d4b-b53ff2e49414.png)

![image](https://cloud.githubusercontent.com/assets/2919859/22407625/6a490fa4-e66a-11e6-8207-f4f261e54661.png)

It returns this error on deploy, where ìmagePosition` is the alias of the fist properties using radiobuttons in this grid editor.

![image](https://cloud.githubusercontent.com/assets/2919859/22407637/a67c5422-e66a-11e6-9dbf-001d90666218.png)

When I select a value for this one and deploy again it returns this error, `textAlignment` is the next properties later in the grid editor:

![image](https://cloud.githubusercontent.com/assets/2919859/22407649/df41adf2-e66a-11e6-9e75-8d299c06c303.png)

If I also select a value for this, it succesfully package the data and is able to deploy the changes.

![image](https://cloud.githubusercontent.com/assets/2919859/22407668/4a300c62-e66b-11e6-93c9-71072987430c.png)

![image](https://cloud.githubusercontent.com/assets/2919859/22407676/65d67ee2-e66b-11e6-8769-5741bdd1b3a8.png)

**Comment by bjarnef** [comment]
The json of the last row with the grid editor, before selecting a value for these two radiobuttonlist properties, looks like the following. So it is probably missing a null check somewhere in the dataresolver.

```
{
    "name": "1 kolonne",
    "areas": [
    {
        "grid": 12,
        "hasConfig": false,
        "controls": [
        {
            "value": [
            {
                "image": {
                "value": "1086",
                "dataTypeGuid": "82958a7c-fffb-47de-a096-a36079108f46",
                "editorAlias": "image",
                "editorName": "Billede"
                },
                "imagePosition": {
                "value": null,
                "dataTypeGuid": "472df8ed-5a4a-4558-96dc-4049ff452889",
                "editorAlias": "imagePosition",
                "editorName": "Placering af billede"
                },
                "heading": {
                "value": null,
                "dataTypeGuid": "0cc0eba1-9960-42c9-bf9b-60e150b429ae",
                "editorAlias": "heading",
                "editorName": "Overskrift"
                },
                "bgColor": {
                "value": null,
                "dataTypeGuid": "85bd0ab5-caa3-45ec-b376-86e83366237d",
                "editorAlias": "bgColor",
                "editorName": "Baggrundsfarve"
                },
                "text": {
                "value": null,
                "dataTypeGuid": "c6bac0dd-4ab9-45b1-8e30-e4b619ee5da3",
                "editorAlias": "text",
                "editorName": "Tekst"
                },
                "textColor": {
                "value": null,
                "dataTypeGuid": "0c6fe886-0667-4a91-9084-d0ee68a9c631",
                "editorAlias": "textColor",
                "editorName": "Tekst farve"
                },
                "textAlignment": {
                "value": null,
                "dataTypeGuid": "472df8ed-5a4a-4558-96dc-4049ff452889",
                "editorAlias": "textAlignment",
                "editorName": "Tekst justering"
                },
                "textSize": {
                "value": null,
                "dataTypeGuid": "0cc0eba1-9960-42c9-bf9b-60e150b429ae",
                "editorAlias": "textSize",
                "editorName": "Tekst størrelse"
                },
                "textOverlay": {
                "value": "0",
                "dataTypeGuid": "92897bc6-a5f3-4ffe-ae27-f2e7e33dda49",
                "editorAlias": "textOverlay",
                "editorName": "Tekst ovenpå billede"
                },
                "link": {
                "value": null,
                "dataTypeGuid": "e27ba7e7-3148-4246-8d13-d0174e0835c8",
                "editorAlias": "link",
                "editorName": "Link"
                }
            }
            ],
            "editor": {
            "alias": "colorBoxWithImage"
            },
            "active": true,
            "guid": "1808c794-1010-93c7-09d7-289e7f4292b4"
        }
        ],
        "active": true
    }
    ],
    "hasConfig": false,
    "id": "04f2542e-e143-63ad-10d7-cce775b9a536",
    "hasActiveChild": true,
    "active": true
}
```

**Comment by paulsterling** [comment]
I don't think it's a null issue but one with "special" characters and Json.  I can repro with a "color picker" radio button list as a LeBlender property.  With no selection (null) the character "¤" is persisted in the Json and the data - and that is what causes this error when packaging.  At least in the repro I have been able to do.
So, not clear at this point if it is an issue with how Courier handles "special" characters when serializing to Json or how the radio button list behaves when used as a LeBlender property.

**Comment by bjarnef** [comment]
@paulsterling I also meant that, it is the value that is null in the json for the property e.g.

```
"imagePosition": {
      "value": null,
      "dataTypeGuid": "472df8ed-5a4a-4558-96dc-4049ff452889",
      "editorAlias": "imagePosition",
      "editorName": "Placering af billede"
}
```

when I prevalue has been selected it looks like this:

```
"imagePosition": {
      "value": "9",
      "dataTypeGuid": "472df8ed-5a4a-4558-96dc-4049ff452889",
      "editorAlias": "imagePosition",
      "editorName": "Placering af billede"
}
```

**Comment by paulsterling** [comment]
@bjarnef Yes, I think we are in agreement...but the value that causes the issue for the json parsing is not the null, but the odd character...and that's the part that we need to figure out where it comes from.  I'm working on that, but no conclusion yet.  Open to suggestions.

**Comment by bjarnef** [comment]
Okay, it just seems like it somewhere might trying to access the value property of null.
A wild guess, could `leBlenderPropertyWrapper.Value` here be null?
https://github.com/umbraco/Umbraco.Courier.Contrib/blob/dev/src/Umbraco.Courier.Contrib.Resolvers/GridCellDataResolvers/LeBlenderGridCellResolver.cs#L56

e.g. is leBlenderPropertyWrapper matching the `imagePosition` json object mentioned before and Value property is `value`. If that is the case, doesn't it then try to use `.ToString()` on null?

**Comment by bjarnef** [comment]
Furthermore isn't `.ToString() as string` unnecessary - when using `.ToString()` it becomes a string a no need to cast it.

**Comment by davidtwamley** [comment]
I just upgraded to 3.0.4 from 2.52.15 and encountered the same error. When I drill into the revision properties folder I have 30 matches for the ¤ character in the json properties. Most are `"value": "¤"` some are `"value": "¤characters here"` ~~the oddest is `"c¤onfig": null`~~ (haven't seen it twice, may have pasted when searching).

**Comment by davidtwamley** [comment]
I don't see the resolver in the stack trace. I started a bug over here too: http://issues.umbraco.org/issue/COU-513

**Comment by davidtwamley** [comment]
I narrowed it down a bit. I can catch it in the act on [this line](https://github.com/umbraco/Umbraco.Courier.Contrib/blob/dev/src/Umbraco.Courier.Contrib.Resolvers/GridCellDataResolvers/LeBlenderGridCellResolver.cs#L93):

`ResolutionManager.Instance.PackagingItem(pseudoPropertyDataItem, propertyDataItemProvider);`

The value of `pseudoPropertyDataItem.Data.First().Value` before is `""` and after line 93 executes it is `"¤"`.

By the way, the `as string` bit is meant to keep it from being a dynamic.

**Comment by davidtwamley** [comment]
I'm seeing this with Umbraco.DropDown based data types used within LeBlender too. It might be entirely on the source side. I'm not catching any traffic to the target server with Fiddler.

**Comment by davidtwamley** [comment]
Had to configure the website to loop through the fiddler proxy. The error is coming from the target server.

**Comment by davidtwamley** [comment]
I setup two rigs and debugged on both sides. I can prevent the transfer error from occurring if I strip out the `¤` after `ResolutionManager.Instance.PackagingItem` is called, but then on extraction it won't resolve it back to an id. If I add back the `¤` just before calling `ResolutionManager.Instance.ExtractingItem` it does resolve it back to an id. So I'm fairly convinced that prefixing with `¤` is an intentional Courier operation that means go look up the matching prevalue id.

**Comment by davidtwamley** [comment]
Got it! There are probably other corner cases that could make for invalid json. I can imagine `"value": 1234` turning into `"value": a3af3895-df1e-4e86-986f-9b2536caf199` for example though I haven't seen it happen yet. Probably need to beef that up even more.

**Comment by bjarnef** [comment]
I just noticed I still had some issues deploying content in latest version of Umbraco and Courier, when no option has been selected in LeBlender radiobutton list.

For this single node it was caused by no option was selected in the radiobuttonlist for this grid editor.

![clipboard](https://cloud.githubusercontent.com/assets/2919859/24492827/f106f736-152c-11e7-8eb9-d6e5b16b33e6.png)

![image](https://cloud.githubusercontent.com/assets/2919859/24492799/def0dc74-152c-11e7-8f0e-7efa47c25d25.png)

After selection of an option I could deploy the node.
